Lib/test/test_winapi.py

```diff
@@ -127,3 +127,37 @@ def test_getshortpathname(self):
 
         # Should contain "PROGRA~" but we can't predict the number
         self.assertIsNotNone(re.match(r".\:\\PROGRA~\d", actual.upper()), actual)
+
+    def test_namedpipe(self):
+        # Deliberately complex name to flush out encoding errors
+        pipe_name = os.urandom(64).replace(b'\0', b'.').decode('latin-1')
+        pipe_name = rf"\\.\pipe\LOCAL\{pipe_name}"
+
+        # Pipe does not exist, so this raises
+        with self.assertRaises(FileNotFoundError):
+            _winapi.WaitNamedPipe(pipe_name, 0)
+
+        pipe = _winapi.CreateNamedPipe(
+            pipe_name,
+            _winapi.PIPE_ACCESS_DUPLEX,
+            8, # 8=PIPE_REJECT_REMOTE_CLIENTS
+            2, # two instances available
+            32, 32, 0, 0)
+        self.addCleanup(_winapi.CloseHandle, pipe)
+
+        # Pipe instance is available, so this passes
+        _winapi.WaitNamedPipe(pipe_name, 0)
```

Comment:
You can also directly check whether the named pipe exists in the directory listing `os.listdir('//./pipe/')`.

Reply:
I can, but then I have to test `WaitNamedPipe` as well :)